File: solutions/java/log-levels/2/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        return logLine.substring(logLine.indexOf(':') + 1).trim();
    }

    public static String logLevel(String logLine) {
        return logLine.substring(logLine.indexOf('[') + 1, logLine.indexOf(']')).toLowerCase();
    }

    public static String reformat(String logLine) {
        return String.join("", message(logLine), " (", logLevel(logLine), ")");
    }
}

File: solutions/cpp/log-levels/2/log_levels.cpp
#include <string>

namespace log_line {
std::string message(std::string line) {
    auto message_begin = line.find("]: ") + 3;
    return line.substr(message_begin);
}

std::string log_level(std::string line) {
    auto level_begin = line.find("[")+1;
    auto level_end = line.find("]");
    auto level_length = level_end - level_begin;
    return line.substr(level_begin, level_length);
}

std::string reformat(std::string line) {
    return message(line) + " (" + log_level(line) + ")";
}
}
